auth/password_authenticator.cc

```diff
@@ -125,12 +132,13 @@ future<> password_authenticator::migrate_legacy_metadata() const {
 }
 
 future<> password_authenticator::legacy_create_default_if_missing() {
-    const auto exists = co_await legacy::default_role_row_satisfies(_qp, &has_salted_hash, _superuser);
+    SCYLLA_ASSERT(_superuser);
```

Comment:
In general, I'm not happy about adding SCYLLA_ASSERT (https://github.com/scylladb/scylladb/issues/7871). In this case, why is _superuser even an "optional" if you plan to assert it must be set?
Or, maybe you'll remove this assert in a a following patch? I'll see...